cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -45,80 +109,302 @@ def evaluate_dynamic_defaults(self, raw_data, **kwargs):
+
+        if throughput_mode == "provisioned" and not provisioned_throughput:
+            raise ValidationError(
+                message="When specifying 'throughput_mode' to 'provisioned',"
+                " the 'provisioned_throughput' option must be specified",
+                field_name="ProvisionedThroughput",
+            )
+
+
+class FsxSchema(BaseSchema):
+    """Represent the schema of FSX."""
+
+    storage_capacity = fields.Int()
+    deployment_type = fields.Str(validate=validate.OneOf(["SCRATCH_1", "SCRATCH_2", "PERSISTENT_1"]))
+    imported_file_chunk_size = fields.Int()
+    export_path = fields.Str()
+    import_path = fields.Str()
+    weekly_maintenance_start_time = fields.Str(validate=validate.Regexp(r"^[1-7]:([01]\d|2[0-3]):([0-5]\d)$"))
+    automatic_backup_retention_days = fields.Int(validate=validate.Range(min=0, max=35))
+    copy_tags_to_backups = fields.Bool()
+    daily_automatic_backup_start_time = fields.Str(validate=validate.Regexp(r"^([01]\d|2[0-3]):([0-5]\d)$"))
+    per_unit_storage_throughput = fields.Int(validate=validate.OneOf(FSX_SSD_THROUGHPUT + FSX_HDD_THROUGHPUT))
+    backup_id = fields.Str(validate=validate.Regexp("^(backup-[0-9a-f]{8,})$"))
+    kms_key_id = fields.Str()
+    file_system_id = fields.Str(validate=validate.Regexp(r"^fs-[0-9a-z]{17}$"))
+    auto_import_policy = fields.Str(validate=validate.OneOf(["NEW", "NEW_CHANGED"]))
+    drive_cache_type = fields.Str(validate=validate.OneOf(["READ"]))
+    storage_type = fields.Str(validate=validate.OneOf(["HDD", "SSD"]))
+
+    @validates("imported_file_chunk_size")
+    def fsx_imported_file_chunk_size_validator(self, value):
+        """Validate ImportedFileChunkSize."""
+        if not 1 <= value <= 512000:
+            raise ValidationError("has a minimum size of 1 MiB, and max size of 512,000 MiB")
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return FsxConfig(**data)
+
+
+class SharedStorageSchema(BaseSchema):
+    """Represent the generic SharedStorage schema."""
+
+    mount_dir = fields.Str(required=True, validate=_get_field_validator("file_path"))
+    ebs_settings = fields.Nested(EbsSchema, data_key="EBS")
+    efs_settings = fields.Nested(EfsSchema, data_key="EFS")
+    fsx_settings = fields.Nested(FsxSchema, data_key="FSX")
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Return the right SharedStorage type object, according to the given type."""
+        return SharedStorageConfig(**data)
+
+    @validates("mount_dir")
+    def validate_not_none_dir(self, value):
+        """Validate that user is not specifying /NONE or NONE as MountDir for any filesystem."""
+        if re.match("^/?NONE$", value):
+            raise ValidationError("{0} cannot be used as a mount directory".format(value))
+
+    @validates_schema
+    def validate_existence_of_settings(self, data, **kwargs):
+        """Validate that there is a setting."""
+        pass
```

Comment:
let's add a # TODO

Reply:
Good catch. Instead of adding TODO, I implemented the method.